streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -697,7 +693,7 @@ private Map<TopicPartition, Long> committedOffsetForChangelogs(final Map<TaskId,
                                                                    final Set<TopicPartition> partitions) {
         final Map<TopicPartition, Long> committedOffsets;
         try {
-            committedOffsets = fetchCommittedOffsets(partitions, mainConsumer);
```

Comment:
I'm thinking that we just ignore the interrupted exception inside the `StoreChangelogReader`, since when it happens, from the `StoreChagenlogReader`'s pov we can always just retry in the next loop, and then if it is due to shutdown we will check the boolean at the beginning of the iteration and shutdown anyways --- i.e. we may be shutdown a bit later, but that is consistent with any other possible interruption scenarios, wdyt?

Reply:
I think re-throwing the `InterruptedException` is cleaner. Otherwise we defeat the purpose of interrupting a thread. Especially, in this shutdown case were we use interruption specifically to interrupt any actions where possible. The old code path should not be affected since we control the stream thread and we never interrupt it.
The only drawback that I see is that we need to add the `throws` expression to the calls in the stacktrace and handle (i.e., ignore) the `InterruptedException` in the old code path in the stream thread.

Reply:
Hi @cadonna I spent some time trying out the idea, but it did not work very elegantly. The main reason is that we have both `committedOffsetForChangelogs` and `endOffsetForChangelogs` which may throw `ExecutionException` / `TimeoutException` and `InterruptedException`. And if we want to let the `InterruptedException` to rethrow, like you said we need to let the caller to handle that, but the current logic is that we would swallow and record the task idleness, so to be compatible we'd still need to do so, which means that we would end up doing similar things for the stream thread.

So I suggest we do what you suggested after we completed moving the store changelog reader to the restore thread, and then change the semantics as "record task idleness for retriable execution and timeout exception, but re-throw for interrupted exception". WDYT?